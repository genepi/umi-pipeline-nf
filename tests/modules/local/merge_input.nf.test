nextflow_process {

    name "Test Process MERGE_FASTQ"
    script "modules/local/umi_processing/merge_input.nf"
    process "MERGE_FASTQ"

    test("Should run without failures: offline") {

        when {
            params {
                min_read_length = 1000
                min_qscore = 9
                live = false
            }
            process {
                """
                input[0] = Channel.of(
                    ["barcode03", 
                        ["$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_0.fastq.gz", 
                        "$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_1.fastq.gz", 
                        "$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_2.fastq.gz"],
                    ]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should run without failures: live") {

        when {
            params {
                min_read_length = 1000
                min_qscore = 9
                live = true
            }
            process {
                """
                input[0] = Channel.of(
                    ["barcode03", "$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_0.fastq.gz"],
                    ["barcode03", "$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_1.fastq.gz"],
                    ["barcode03", "$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_2.fastq.gz"],
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            
        }

    }

    test("Should output empty file") {

        when {
            params {
                min_read_length = 10000
                min_qscore = 9
            }
            process {
                """
                input[0] = Channel.of(
                    ["barcode03", 
                        ["$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_0.fastq.gz", 
                        "$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_1.fastq.gz", 
                        "$baseDir/tests/input/merge_fastq/fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_2.fastq.gz"],
                    ]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
