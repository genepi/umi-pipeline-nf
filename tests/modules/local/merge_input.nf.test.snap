{
    "Should run without failures: live": {
        "content": [
            {
                "0": [
                    [
                        "barcode03",
                        "target",
                        "filtered_fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_0.fastq:md5,59011fb010a30b71e9db62fc880b508e"
                    ],
                    [
                        "barcode03",
                        "target",
                        "filtered_fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_1.fastq:md5,8ef8fdc7328409cfc34e5036f8023d7b"
                    ],
                    [
                        "barcode03",
                        "target",
                        "filtered_fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_2.fastq:md5,420a9ed6574702fa025b2580315a3d1c"
                    ]
                ],
                "merged_fastq": [
                    [
                        "barcode03",
                        "target",
                        "filtered_fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_0.fastq:md5,59011fb010a30b71e9db62fc880b508e"
                    ],
                    [
                        "barcode03",
                        "target",
                        "filtered_fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_1.fastq:md5,8ef8fdc7328409cfc34e5036f8023d7b"
                    ],
                    [
                        "barcode03",
                        "target",
                        "filtered_fastq_runid_b05e9ba03fee394fc76a041d3df7426ff48616c9_2.fastq:md5,420a9ed6574702fa025b2580315a3d1c"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-18T17:39:52.858568"
    },
    "Should output empty file": {
        "content": [
            {
                "0": [
                    [
                        "barcode03",
                        "target",
                        "barcode03_merged.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "merged_fastq": [
                    [
                        "barcode03",
                        "target",
                        "barcode03_merged.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-18T17:40:57.970231"
    },
    "Should run without failures: offline": {
        "content": [
            {
                "0": [
                    [
                        "barcode03",
                        "target",
                        "barcode03_merged.fastq:md5,1a65a32bc2d389105716ebebf13a3d40"
                    ]
                ],
                "merged_fastq": [
                    [
                        "barcode03",
                        "target",
                        "barcode03_merged.fastq:md5,1a65a32bc2d389105716ebebf13a3d40"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-18T17:39:37.774507"
    }
}